web/src/components/common/ListingCollection.tsx

```diff
@@ -44,44 +44,41 @@ const ListingItem: React.FC<ListingProps> = ({
     minCommits,
     imgUrl,
   },
-}) => (
-  <StrippedCol xs={6} key={id}>
-    <Link
-      to={{
-        pathname: `/listing/${id}`,
-        state: {
-          fromExplore: true,
-        },
-      }}
-    >
-      <ListingCard
-        listingName={name}
-        price={price}
-        oldPrice={oldPrice}
-        endDate={deadline}
-        imgUrl={imgUrl}
+}) => {
+  const isMobile = useMediaQuery({query: '(max-device-width: 768px)'});
```

Comment:
can you move this to a common place so that it could be used by other places?
also this check looks not straightforward, is there other way like checking the user agent?

Reply:
@ifhuang Okay! I will put this in a context like @hzxcaryn suggested. 
Hmm I don't know of a more straightforward way. But I think usually people use media query to check the display size, and then format the content accordingly.

Reply:
@ifhuang I think it is better to use media query than checking user agent. 
I think checking user agent works for examples like whether clicking a button would lead them to app store, or checking if sharing a link via sms is a valid button. 
However, what Karen wants to achieve here is to alter the layout of the page, depending on the screen width. This `isMobile` here is not literally checking for whether the user is using a mobile or not, but whether the screen width is that of a mobile phone.
I think that we should stick with using media queries for "styling logic", and refactor using the suggestions I wrote in my [review below](https://github.com/googleinterns/gpay-group-buy/pull/141#pullrequestreview-452110650).

Key reason for this is that not all people using their laptops will view the page in full screen. They might split screen (or open developer inspect) and the page would appear pretty narrow. In cases like these we would actually prefer the listings to be wider instead of squashing the listings simply because they are not browsing using a mobile phone.

This is what is happening to the page now if I do that on laptop:
![image](https://user-images.githubusercontent.com/25261058/88050586-b4a6a680-cb89-11ea-8762-7429ca751d42.png)

Reply:
Thanks for the detailed comment.
I think your example with split screen on the laptop is a valid use use. In terms of var name, we should probably use sth. like "isSmallScreen" instead of "isMobile".

Reply:
@hzxcaryn That's a nice corner case that supports the use of media query. I agree that media query is more relevant to screen sizes which we are concerned about here. I didn't realise that react-device-detect uses User Agent and wasn't really testing for responsiveness just now, since we are prioritising on the main cases for now. I thought that react-device-detect works for now since it looks cleaner on the code and is immediately understandable, although it doesn't deal with the corner cases.

Since this is related to responsiveness, I will add this particular case to this issue #112 (Make UI responsive).

@ifhuang Okay! That is probably better to avoid this kind of misunderstanding.